lib/perl/Genome/Utility/IO/SeparatedValueReader.pm

```diff
@@ -89,7 +91,21 @@ sub create {
     else {
         $regexp = qr/\Q$sep\E/;
     }
-    $self->{_split} = sub{ return _strip_quotes(split($regexp, $_[0], -1)) };
+
+    if ($sep eq ',') {
+        my $csv = Text::CSV->new({ binary => 1, auto_diag => 1, sep_char => ',' });
+        $self->{_split} = sub {
+            my $line = $_[0];
+            chomp($line);
+            if ($csv->parse($line)) {
+                return _strip_quotes($csv->fields());
+            } else {
+                $self->fatal_message('Failed to parse CSV line: '. $line);
+            }
+        };
+    } else {
+        $self->{_split} = sub{ return _strip_quotes(split($regexp, $_[0], -1)) };
+    }
```

Comment:
I agree Text::CSV could be used more in this module.  I was focused on adding support of true CSV parsing that uses text delimiters in addition to the field delimiter.

As for turning each entry into a UR object, I believe this is why some parts of our codebase still use separated value reader instead of filesystem data sources.  It's much faster to read millions of lines when simple hash refs are returned rather than loading real objects for each line of text.  However, I could imagine a single interface that would allow the optional creation of real objects for each line.